### PR TITLE
[Feature] Separate download & update player from launcher update

### PR DESCRIPTION
**Pull request by Atralupus**
Solved #1704

# Description
All tasks for `Separate player` are merged in this huge PR.

**First major update** is `Doesn't import downloadUrl from APV version`
The `DownloadURL` is created according to the version scheme, and the `DownloadURL` attempts to update it assuming it exists.

**Secondly major update** is you will no longer install the player in the same folder as Launcher.
The update script has been detached to be installed separately in the [userData folder](https://www.electronjs.org/docs/latest/api/app#appgetpathname) and accordingly, modified player execute path.

And... **minor updates**
- If there is no player under the userData folder for the first time, start `player update`
- Not proceeding bundle-player in CI.

This PR will not be merged until the [Internal test](https://github.com/planetarium/9c-launcher/issues/1760) is complete.

### TODO
- [x] Separate update player
- [x] Download URL problem
- [x] ~~Remove 9c submodule~~
- [x] Fix circle ci
- [x] Update downloadPlayer
---
Merged PR
- https://github.com/planetarium/9c-launcher/pull/1727
- https://github.com/planetarium/9c-launcher/pull/1738
- https://github.com/planetarium/9c-launcher/pull/1745
- https://github.com/planetarium/9c-launcher/pull/1761
- https://github.com/planetarium/9c-launcher/pull/1767
- https://github.com/planetarium/9c-launcher/pull/1784

### CODE REVIEW

**Comment by Atralupus** [comment]
TODO @Atralupus Check this [comment](https://github.com/planetarium/9c-launcher/pull/1727#pullrequestreview-1067542567)

**Comment by Atralupus** [comment]
Remove 9c submodule
^ 9c submodule exists only to obtain a commit hash, so I'll not remove it in this PR

**Comment by Atralupus** [comment]
> TODO @Atralupus Check this [comment](https://github.com/planetarium/9c-launcher/pull/1727#pullrequestreview-1067542567)

Done

**Comment by Atralupus** [comment]
@Atralupus Check https://github.com/planetarium/9c-launcher/pull/1790#discussion_r949900821

File: src/v2/machines/updateMachine.ts [hunk 1]
```diff
@@ -23,76 +24,121 @@ type UpdateMachineState =
...
-                "update download started",
-                "DOWNLOAD"
-              ),
-          },
-          {
-            id: "triggerUpdate",
-            src: () => ipcRenderer.invoke("start update"),
-          },
-        ],
+    extract: {
```
Comment (BasixKOR) [inline src/v2/machines/updateMachine.ts:46]: Missing `UPDATE_PROGRESS` handler.

File: src/v2/machines/updateMachine.ts [hunk 2]
```diff
@@ -23,76 +24,121 @@ type UpdateMachineState =
...
-      updateProgress: assign({
-        progress: (context, event) =>
-          event.type === "UPDATE_PROGRESS" ? event.progress : context.progress,
-      }),
+    playerUpdate: {
+      ...playerUpdate,
+    },
+    launcherUpdate: {
+      ...launcherUpdate,
     },
```
Comment (BasixKOR) [inline src/v2/machines/updateMachine.ts:142]: Can be shortened down to:

```javascript
{
  playerUpdate,
  launcherUpdate,
}
```

**Review by BasixKOR — APPROVED** [review]

**Comment by Atralupus** [comment]
:rocket: :rocket: :rocket: :rocket: :rocket:

**Review by Akamig — APPROVED** [review]

**Pull request merged**